mentat/terminal/patched_autocomplete.py

```diff
@@ -0,0 +1,191 @@
+from dataclasses import dataclass
+from typing import List
+
+from textual import events
+from textual_autocomplete import AutoComplete, Dropdown, DropdownItem
+from typing_extensions import override
+
+from mentat.auto_completer import Completion
+from mentat.session_stream import SessionStream, StreamMessageSource
+from mentat.terminal.history_suggester import HistorySuggester
+
+
+class PatchedAutoComplete(AutoComplete):
+    """
+    We want more fine control over replacing the text when a completion is
+    selected, so we patch _select_item.
+    """
+
+    @override
+    def _select_item(self):
+        selected: CompletionDropdownItem | None = (
+            self.dropdown.selected_item
+        )  # pyright: ignore
+        if self.dropdown.display and selected is not None:
+            cursor_position: int = self.input.cursor_position  # pyright: ignore
+            before = self.input.value[: cursor_position + selected.position]
+            after = self.input.value[cursor_position:]
+            self.input.value = before + selected.content + after
+            self.input.cursor_position = (
+                cursor_position + selected.position + len(selected.content)
+            )
+
+            self.dropdown.display = False
+            self.post_message(self.Selected(item=selected))
+
+
+@dataclass
+class CompletionDropdownItem(DropdownItem):
+    """
+    A dropdown item with a full completion to carry all of the information we need
+    """
+
+    content: str = ""
+    position: int = 0
+
+
+class PatchedDropdown(Dropdown):
```

Comment:
We can't change the text size (since we're working in a terminal), so I don't think this would end up looking very good. Maybe a different key would be better than tab?

Reply:
could it just be Enter? that's what I do instinctually, and it's a lot of work to recover.